src/subdomains/supporting/dex/services/dex-monero.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@nestjs/common';
+import { Config } from 'src/config/config';
+import { MoneroTransactionType, MoneroTransferDto } from 'src/integration/blockchain/monero/dto/monero.dto';
+import { MoneroClient } from 'src/integration/blockchain/monero/monero-client';
+import { MoneroHelper } from 'src/integration/blockchain/monero/monero-helper';
+import { MoneroService } from 'src/integration/blockchain/monero/services/monero.service';
+import { Blockchain } from 'src/integration/blockchain/shared/enums/blockchain.enum';
+import { Util } from 'src/shared/utils/util';
+import { LiquidityOrder } from '../entities/liquidity-order.entity';
+import { LiquidityOrderRepository } from '../repositories/liquidity-order.repository';
+
+@Injectable()
+export class DexMoneroService {
+  private client: MoneroClient;
+
+  constructor(private readonly liquidityOrderRepo: LiquidityOrderRepository, moneroService: MoneroService) {
+    this.client = moneroService.getDefaultClient();
+  }
+
+  async sendTransfer(address: string, amount: number): Promise<string> {
+    return this.client.sendTransfer(address, amount).then((r) => r.txid);
+  }
+
+  async transferMinimal(address: string): Promise<string> {
+    return this.sendTransfer(address, Config.payIn.minDeposit.Monero.XMR / 2);
+  }
+
+  async checkTransferCompletion(transferTxId: string): Promise<boolean> {
+    const transaction = await this.client.getTransaction(transferTxId);
+
+    return MoneroHelper.isTransactionComplete(transaction);
+  }
+
+  async getRecentHistory(blockCount: number): Promise<MoneroTransferDto[]> {
+    const currentBlockHeight = await this.client.getBlockHeight();
+    return this.client.getTransfers(MoneroTransactionType.in, currentBlockHeight - blockCount);
+  }
+
+  async checkAvailableTargetLiquidity(inputAmount: number): Promise<[number, number]> {
+    const pendingAmount = await this.getPendingAmount();
+    const availableAmount = await this.client.getBalance().then((b) => b.unlocked_balance);
+
+    return [inputAmount, availableAmount - pendingAmount];
+  }

```

Comment:
Shouldn't use the unlocked_balance but the complete one. Otherwise the liquidity management will get more, even though there is enough.